### PR TITLE
Enhancement #16699: Designer warns when creating relationships

### LINKED ISSUE #16699: Designer should warn when trying to create relationships of different data types
### Is your feature request related to a problem? Please describe.

When using Designer, it's easy to accidentally try to create a relationship between different data types, for instance a TINYINT referenced key with an INT foreign key. The MySQL error message is not very helpful to figuring out what the problem is.

### Describe the solution you'd like

phpMyAdmin could warn the user when they're attempting to add a relationship between data types that are different; we already have that information for display in Designer it's just a matter of comparing them and showing the warning. I think a warning should have an OK and Cancel button prompting the user if they want to proceed.

### Describe alternatives you've considered

We could just tell the user "you can't do that because they're different data types" and prohibit the action, but I think the dialog is better because as a general rule, we don't stop the user from doing things, we just warn them first.

### Additional context

- phpMyAdmin version: 5.1.0

### PR DESCRIPTION
**Pull request by shucon**
Designer warns when creating relationships

Signed-off-by: Saksham Gupta <shucon01@gmail.com>

### Description

Please describe your pull request.

Fixes #16699 

Before submitting pull request, please review the following checklist:

- [ ] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [ ] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [ ] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [ ] Every commit has a descriptive commit message.
- [ ] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [ ] Any new functionality is covered by tests.

### CODE REVIEW

File: js/src/designer/move.js [hunk 1]
```diff
@@ -1280,17 +1290,20 @@ DesignerMove.clickField = function (db, T, f, pk) {
...
             }
-            var left = globX - (document.getElementById('layer_new_relation').offsetWidth >> 1);
-            document.getElementById('layer_new_relation').style.left = left + 'px';
-            var top = globY - document.getElementById('layer_new_relation').offsetHeight;
-            document.getElementById('layer_new_relation').style.top  = top + 'px';
-            document.getElementById('layer_new_relation').style.display = 'block';
+            prevX = globX;
+            prevY = globY;
+            if (fieldType !== type) {
+                DesignerMove.displayTable('layer_cnfrm_relation');
```
Comment (mauriciofauth) [inline js/src/designer/move.js:1303]: What does `cnfrm` mean? Please don't use this kind of abbreviations.

**Review by MauricioFauth — CHANGES_REQUESTED** [review]

File: js/src/designer/move.js [hunk 2]
```diff
@@ -1264,8 +1267,15 @@ DesignerMove.startRelation = function () {
     }
 };
 
+DesignerMove.displayTable = function (name) {
+    var left = prevX - (document.getElementById(name).offsetWidth >> 1);
```
Comment (williamdes) [inline js/src/designer/move.js:1271]: `document.getElementById(name)` can be stored in a variable

File: js/src/designer/move.js [hunk 3]
```diff
@@ -2236,5 +2251,12 @@ AJAX.registerOnload('designer/move.js', function () {
     $('input#cancel_new_rel_panel').on('click', function () {
         document.getElementById('layer_new_relation').style.display = 'none';
     });
+    $('input#cnfrm_rel').on('click', function () {
```
Comment (williamdes) [inline js/src/designer/move.js:2254]: Same here for `cnfrm`

File: js/src/designer/move.js [hunk 4]
```diff
@@ -2236,5 +2251,12 @@ AJAX.registerOnload('designer/move.js', function () {
...
+        document.getElementById('layer_confirm_relation').style.display = 'none';
```
Comment (williamdes) [inline js/src/designer/move.js:2255]: I think there is a better Jquery way to do this 🤔

File: templates/database/designer/main.twig [hunk 5]
```diff
@@ -359,6 +359,60 @@ var designerConfig = {{ designer_config|raw }};
...
+    <td class="frams1" width="10px">
+    </td>
+    <td class="frams5" width="99%" >
+    </td>
+    <td class="frams2" width="10px">
+      <div class="bor">
+      </div>
+    </td>
+  </tr>
+  <tr>
```
Comment (williamdes) [inline templates/database/designer/main.twig:375]: this feels a bit strange

Comment (shucon) [reply templates/database/designer/main.twig:375]: The template is copied from other tables.

File: js/src/designer/move.js [hunk 6]
```diff
@@ -2236,5 +2252,12 @@ AJAX.registerOnload('designer/move.js', function () {
     $('input#cancel_new_rel_panel').on('click', function () {
         document.getElementById('layer_new_relation').style.display = 'none';
     });
+    $('input#confirm_rel').on('click', function () {
+        $('#layer_confirm_relation').hide();
+        DesignerMove.displayTable('layer_new_relation');
+    });
+    $('input#cancel_rel').on('click', function () {
+        document.getElementById('layer_confirm_relation').style.display = 'none';
```
Comment (williamdes) [inline js/src/designer/move.js:2260]: this should be jquery

**Review by williamdes — CHANGES_REQUESTED** [review]
I think there is a better way to make this feature and including more use cases like data types, feel free to open a new one

**Pull request closed without merging**